### ISSUE TITLE
Suggestion for Improvements in Export to Vegas EDL Option

### ISSUE THREAD

**Issue by Joelson-Forte** [protyposis/AudioAlign#24]
Firstly, I want to thank you for your work. I did some tests with your app and had good results.

I would like to make a suggestion for improvement in relation to the Export to Vegas EDL option. I noticed that the EDL file creates a video track and an audio track for each clip analyzed.

Let's use a hypothetical situation as an example: Let's say I made a recording with 2 cameras (Camera 01 and Camera 02) and an External Recorder (External Audio)

Camera 01 recorded 10 clips and the clips will be in the "Cam_01" folder. Camera 02 recorded 8 clips and the clips are in the "Cam_02" folder. The external audio will be in the "External Audio" Folder.

When I use AudioAlign on these files and export the EDL file to use in Vegas Pro, there will be 37 tracks in the Vegas Pro Timeline (10 video tracks + 10 Audio Tracks from Camera 01 + 8 Video Tracks + 8 Audio Tracks from Camera 01). camera 02 + 1 audio track from the external recorder) Totaling 37 Tracks.

My suggestion is that there may be a way to export the EDL file while keeping the clips from each folder on a single track.

For example:

1.The 10 Clips from camera 01 will only be on tracks 1 and 2 (track 1 being the 10 video clips and track 2 being the 10 audio clips).

2.The 8 Clips from camera 02 will only be on tracks 3 and 4 (track 3 being the 8 video clips and track 4 being the 8 audio clips).

3.The audio from the external recorder will be on track 5

This way, only 5 Tracks will be created instead of 37, making it easier to locate and organize files on the timeline.

I'm currently editing the EDL file to import clips with this track format into the Vegas timeline to work in Multicam mode.

I made a screen recording showing the entire procedure and I'm also sending all the files and the project I used in the test if you need it.

You can download it from this link below.
https://drive.google.com/file/d/1SmjmlFyubOj3OyZoTcoBUAq_mR8DVYor/view?usp=sharing

If you have any questions, you can contact me. Congratulations for your work!

Thanks!

**Comment by MarcoRavich** [comment]
Yes, it would be better to put absolute paths (eg. C:\Users\Director\Videos\2024\Concert\CMAN\00001.MP4) to avoid EDL manual editing...

...and some file formats (recently added MTS) aren't "mapped" correctly in EDLs 'cause Vegas loads only audio stream but not corresponding video one.

**Comment by protyposis** [comment]
Thank you for this greatly described and documented issue description. I understand the issue and this is certainly something that could be improved. I am not sure yet whether AudioAlign is really the right place to do this kind of clip-to-track grouping, since AA itself does not support this NLE paradigm. I'll keep an eye on this, though.

In the meantime, and as an alternative to manual EDL file editing, the workaround that I use myself is moving the clips to the desired tracks directly within Vegas, either by mouse with the snapping helper (see video below), or even simpler, using the "Move selected events up/down one track" keyboard shortcut `Numeric Keypad 8/2`:

https://github.com/protyposis/AudioAlign/assets/189372/8be2bcda-1d97-422c-a96d-c735da90e918

> it would be better to put absolute paths

AudioAlign uses relative paths to allow portability when the EDL is saved to a parent directory of the media files. When saved to any other location, it uses absolute paths. So if you need absolute paths, save it anywhere else. If you still think this needs to be changed, please open a separate issue.

> some file formats (recently added MTS) aren't "mapped" correctly in EDLs

Will be tracked separately in https://github.com/protyposis/AudioAlign/issues/25.

**Comment by JeditPro** [comment]
> I am not sure yet whether AudioAlign is really the right place to do this kind of clip-to-track grouping, since AA itself does not support this NLE paradigm. I'll keep an eye on this, though.

@protyposis 
Thanks for your answer. I like the way Audioalign works and I think it should stay as is as it works well. I just wish there was a way to generate an EDL file that organized clips from the same camera into a single track.

It could even be a Vegas script that converts the original EDL file to this format.

Unfortunately I don't have much knowledge with scripts, but I got in touch with a friend who is one of the best Vegas Pro scripters I know. If he manages to do this, I'll come back here and share the script with you so that all AudioAlign users can benefit from it.
